File: models/Gpt-3.5-turbo/py/code/sorting_and_searching/Apartments.py

# Read input
n, m, k = map(int, input().split())
desired_sizes = list(map(int, input().split()))
apartment_sizes = list(map(int, input().split()))

# Sort both lists
desired_sizes.sort()
apartment_sizes.sort()

# Initialize variables
i = 0  # Pointer for desired_sizes
j = 0  # Pointer for apartment_sizes
count = 0  # Counter for the number of applicants who will get an apartment

# Iterate through both lists
while i < n and j < m:
    # Check if the current apartment size is within the desired range
    if abs(desired_sizes[i] - apartment_sizes[j]) <= k:
        count += 1
        i += 1
        j += 1
    # If the current apartment size is too small, move to the next apartment size
    elif desired_sizes[i] - apartment_sizes[j] > k:
        j += 1
    # If the current apartment size is too large, move to the next desired size
    else:
        i += 1

# Print the result
print(count)

File: models/Gpt-3.5-turbo/cpp/code/problems/sorting_and_searching/Apartments.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n, m, k;
    cin >> n >> m >> k;

    vector<int> applicants(n);
    for (int i = 0; i < n; i++) {
        cin >> applicants[i];
    }

    vector<int> apartments(m);
    for (int i = 0; i < m; i++) {
        cin >> apartments[i];
    }

    sort(applicants.begin(), applicants.end());
    sort(apartments.begin(), apartments.end());

    int count = 0;
    int i = 0, j = 0;
    while (i < n && j < m) {
        if (abs(applicants[i] - apartments[j]) <= k) {
            count++;
            i++;
            j++;
        } else if (applicants[i] < apartments[j]) {
            i++;
        } else {
            j++;
        }
    }

    cout << count << endl;

    return 0;
}
